src/sys.d.ts

```diff
@@ -0,0 +1,1352 @@
+export type JsObj = Obj | number | string | boolean | ((...args: any) => any)
+
+export class Obj {
+  equals(that: JsObj | null): boolean
+  compare(that: JsObj): number
+  hash(): number
+  toStr(): string
+  trap(name$: string, args?: List<JsObj | null> | null): JsObj | null
+  with$(f: ((argA: this) => void)): this
+  isImmutable(): boolean
+  toImmutable(): this
+  typeof$(): Type
+  static echo(x?: JsObj | null): void
+}
+export class Facet extends Obj {
+}
+export class TimeZone extends Obj {
+  static listNames(): List<string>
+  static listFullNames(): List<string>
+  static fromStr(name$: string, checked?: boolean): TimeZone
+  static utc(): TimeZone
+  static rel(): TimeZone
+  static cur(): TimeZone
+  static defVal(): TimeZone
+  name$(): string
+  fullName(): string
+  offset(year: number): Duration
+  dstOffset(year: number): Duration | null
```

Comment:
In general, as commented on the JS code gen, it would be preffered to use `undefined` rather than `null` so this would be
`: Durration | undefined`

Reply:
I'm curious about that - is that a convention TypeScript API use to prefer undefined to null for nullable return types?

Reply:
Yes, this is the convention. It makes the idiosyncratic rules of JS more palatable.

Values in JS are:
1. undefined -> uninitialized data
2. null -> initialized but with a null (none) value
3. value -> initialized with a value

For function returns and argument option 2 makes little sense, and complicates the semantics. I would argue that the second case is to be used rarely in general...